### ISSUE TITLE
move to self hosting

### ISSUE THREAD

**Issue by devhindo** [devhindo/x#62]
fully self hosted - no db - no authentication server
every user should create an app and use it.

due to 17 requests per day per app dummy pricing API plan
- [ ] provide documentations for self-hosting
- [ ] I imagine something like x init and starts a TUI operation to start the initial operations
- [ ] ability to change the default location of the config file - make it in JSON format `.x.json`

**Comment by mesb1** [comment]
17 requests?

I do remember we were discussing 50 isn't it?

**Comment by devhindo** [comment]
@mesb1 Elon keeps reducing it, that's why I suggested a self-hosting solution. For every user they would register their X app and manage their own hosting and have their own plan. But that would require creating a new project lol cause the current one was built counter the idea of self-hosting and it was never been considered while coding it.

**Comment by devhindo** [comment]
it's also fairly easy if you know enough about OAuth 2.0
just a few REST operations

**Comment by mesb1** [comment]
> Elon keeps reducing it

Huh, i will try to reduce my posting via your app.

**Comment by mesb1** [comment]
> For every user they would register their X app and manage their own hosting and have their own plan. But that would require creating a new project lol cause the current one was built counter the idea of self-hosting and it was never been considered while coding it.

Using own vps isn't big deal, if your app self contain everything and just needs to be started for x86\x86-64\arm machines.
Mostly i am worried about steps related for all this steps related to registrations etc.

**Comment by mesb1** [comment]
> require creating a new project

So do you have any plans?

**Comment by devhindo** [comment]
It doesn't really need a vps. it's just a program that executes when you try to post a tweet. making a request to the API and the Access token is stored locally somewhere in the config files. But it just needs some work and I'm not currently free.

**Comment by devhindo** [comment]
I have plans for later yeah.

**Comment by devhindo** [comment]
yeah it's mostly the registration part the requires that OAuth 2.0 knowledge and after that it's just 2-3 requests to the X API
